tools/shared.py

```diff
@@ -3090,7 +3106,8 @@ def add_emscripten_metadata(js_file, wasm_file):
       WebAssembly.lebify(global_base) +
       WebAssembly.lebify(dynamic_base) +
       WebAssembly.lebify(dynamictop_ptr) +
-      WebAssembly.lebify(tempdouble_ptr)
+      WebAssembly.lebify(tempdouble_ptr) +
+      WebAssembly.lebify(int(Settings.STANDALONE_WASM))
```

Comment:
I wonder if we should just not add this metadata for STANDALONE_WASM?  Since the metadata was mostly to support for loading of emscripten-like wasm binaries outside of emscripten?   Perhaps we can do that as a followup after consulting with the users of the metadata?  Who are the users exactly I'm not sure.

Reply:
I think we can check with people if it's unnecessary, but we should either land this with the change (which is the safe thing) or block this until we decide whether to add this change or not. I lean towards the former since the latter may take a while.

In general I think this may still be useful for the same people for the same reasons as before - these wasms are as standalone as we can make them, but still may contain e.g. WebGL calls if using OpenGL, etc. So they are not pure wasi in general. Which means it's useful to have metadata about the emscripten ABI, and this flag does affect the ABI.

Reply:
Sure we can land this now.. I was just suggesting this as a followup.

Reply:
Hmm, I remembered that we have promised to only append to the metadata list. So if we land this we probably don't want to remove the field later.

However, as I said earlier, I think we do want this field, so I'm not worried. Are you still concerned?

Reply:
Re-reading your posts, were you saying that we should not emit `EMSCRIPTEN_METADATA` at all for `STANDALONE_WASM`? (I initially understood you to mean this new field specifically.)

Reply:
I'm not talking about the field, but the entire metadata section.   Its seems to be that the point of this section is so that runtimes and extract the information needs from the "emscripten-flavor" wasm file and run it without JS. 

My idea is that for PURE_WASM we may not need the section at all in the future.  In which case .. this field would be zeor by definition...  or we could bump the major version which allows is the change it however we like.

My real long term hope is that the users of this metadata section are in fact that exact same users who want to use PURE_WASM instead and that PURE_WASM don't needs to self-describe in this way, and so we may be able to simply remove this completely?

Reply:
I see now, thanks.

Yeah, maybe we eventually won't need this metadata at all, if we only use standard APIs. That seems like the very far future though, as we'll need nonstandard APIs to do many things for quite some time, if not forever (e.g. WebGL).